src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { countTodos } from '../../utils/countTodos';
+import { useTodos } from '../Contexts/TodosContext';
+import { useErrorMessage } from '../Contexts/ErrorMessageContext';
+import { USER_ID } from '../../utils/userToken';
+import { useLoadingTodos } from '../Contexts/LoadingTodosContext';
+import { addTodo, updateTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onWaiting: (tempTodo: Todo | null) => void,
+};
+
+export const NewTodo: React.FC<Props> = ({ onWaiting }) => {
+  const { todos, setTodos } = useTodos();
+  const { setLoadingTodos } = useLoadingTodos();
+
+  const [isAdding, setIsAdding] = useState(false);
+
+  const { handleShowError } = useErrorMessage();
+  const [newTitle, setNewTitle] = useState('');
+
+  const uncompletedTodos = countTodos(todos, false);
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!newTitle.trim()) {
+      handleShowError('Title can\'t be empty');
+
+      return;
+    }
+
+    const tempTodo = {
+      id: 0,
+      completed: false,
+      title: newTitle.trim(),
+      userId: USER_ID,
+    };
+
+    onWaiting(tempTodo);
+
+    setIsAdding(true);
+    try {
+      const newTodo = await addTodo(USER_ID, tempTodo);
+
+      setTodos(prev => [...prev, newTodo]);
+      setNewTitle('');
+    } catch {
+      handleShowError('Unable to add todo');
+    }
+
+    onWaiting(null);
+    setIsAdding(false);
+  };
+
+  const completeAll = async () => {
+    const changingTodos = uncompletedTodos.length
+      ? uncompletedTodos
+      : todos;
+
+    const todosPromises = changingTodos.map(async ({ id, completed }) => {
+      return updateTodo(id, { completed: !completed });
+    });
+
+    try {
+      changingTodos.forEach(({ id }) => {
+        setLoadingTodos(prev => [...prev, id]);
+      });
```

Comment:
You add one todoId in every iteration, you can create the ids array with `map` method and pass them to `setLoadingTodos`